### PR TITLE
Initial Profile Page, Post Deletion

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- Initial Profile Page UI
- Post Deletion of own post function and UI

**SNIPPETS**
Delete

### CODE REVIEW

File: src/app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -44,4 +45,13 @@ public function viewAllPosts(): View
         $posts = $this->postService->viewAllPosts();
         return view('/app', ['posts' => $posts]);
     }
+
+    /**
+     * Delete post
+     */
+    public function delete(Post $post): RedirectResponse
+    {
+        $post->delete();
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostController.php]: transfer this to service

File: src/resources/views/pages/dashboard/post-content.blade.php [hunk 2]
```diff
@@ -5,25 +5,54 @@
...
+  function openNav(postId) {
+    document.getElementById("mySidenav" + postId).style.width = "100px";
+    document.getElementById("filter" + postId).style.display = "block";
+  }
+
+  function closeNav(postId) {
+    document.getElementById("mySidenav" + postId).style.width = "0";
+    document.getElementById("filter" + postId).style.display = "none";
+  }
+</script>
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-content.blade.php:57]: You already declared this in post.js.

Comment (PeterDMadrid) [reply src/resources/views/pages/dashboard/post-content.blade.php:57]: Done

File: src/app/Http/Controllers/PostController.php (hunk 1 above)
Comment (PeterDMadrid) [reply src/app/Http/Controllers/PostController.php]: Done

File: src/routes/web.php [hunk 3]
```diff
@@ -30,3 +30,5 @@
 
 //Posts Routes
 Route::post('/post', [PostController::class, 'create']);
+//Delete Posts
```
Comment (mikco-hipe) [inline src/routes/web.php]: I noticed you're not using middlewares on these routes (Post and Delete) does this mean that even unauthenticated users can post and delete posts?

Comment (PeterDMadrid) [reply src/routes/web.php]: I added the post and delete route to the middleware group as the user needs to be authenticated to be able to create a post and delete.

**Review by sunkenquest — APPROVED** [review]
LGTM 🙇

**Pull request merged**